lib/puppet/provider/foreman_global_parameter/rest_v3.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+Puppet::Type.type(:foreman_global_parameter)
+            .provide(
+              :rest_v3,
+              parent: Puppet::Type.type(:foreman_resource).provider(:rest_v3)
+            ) do
+  confine feature: %i[json oauth]
+
+  mk_resource_methods
+
+  # Note that #request is an instance method and as such prevents
+  # self.instances and self.prefetch from being implimented.
+
+  # The foreman api has both `hidden_value` and `hidden_value?` fields. The
+  # later is the boolean value.
+  def hidden_value
+    @property_hash[:hidden_value?]
+  end
+
+  def hidden_value=(value)
+    @property_hash[:hidden_value?] = value
+  end
+
+  def create
+    @property_hash[:ensure] = :present
+  end
+
+  def exists?
+    get_common_parameter!
+    # debug("exists? @property_hash: #{@property_hash}")
+    # debug("exists? resource: #{resource.to_hash}")
+    @property_hash[:ensure] == :present
+  end
+
+  def destroy
+    @property_hash[:ensure] = :absent
+  end
+
+  def flush
+    debug("flush @property_hash: #{@property_hash}")
+
+    case self.ensure
+    when :present
+      # if there is already an id, this is an update
+      if id.nil?
+        create_common_parameter
+      else
+        update_common_parameter
+      end
+    when :absent
+      destroy_common_parameter
+    else
+      raise Puppet::Error, "invalid :ensure value: #{self.ensure}"
+    end
+  end
+
+  private
+
+  def id
+    @property_hash[:id]
+  end
+
+  def base_path
+    'api/v2/common_parameters'
+  end
+
+  def get_common_parameter!
+    debug('Getting')
```

Comment:
Could you remove any of the debug statements that were only needed during development, or at least make them more descriptive than this?

Reply:
I've removed all the debug statement.